imblearn/over_sampling/_rose.py

```diff
@@ -0,0 +1,210 @@
+"""Class to perform over-sampling using ROSE."""
+
+import numpy as np
+from scipy import sparse
+from sklearn.utils import check_random_state
+from .base import BaseOverSampler
+from ..utils._validation import _deprecate_positional_args
+
+
+class ROSE(BaseOverSampler):
+
+    """Random OverSampling Examples (ROSE).
+
+    The algorithm generates new samples by a smoothed bootstrap approach.
+    The generation of new examples corresponds to the generation of data from
+    the kernel density estimate of f(x|Y_i), with a smoothing matrix H_j.
+    A shrinking matrix can be provided, to set the bandwidth of the gaussian
+    kernel.
+
+    Read more in the :ref:`User Guide <rose>`.
+
+    Parameters
+    ----------
+    sampling_strategy : float, str, dict or callable, default='auto'
+        Sampling information to resample the data set.
+
+        - When ``float``, it corresponds to the desired ratio of the number of
+          samples in the minority class over the number of samples in the
+          majority class after resampling. Therefore, the ratio is expressed as
+          :math:`\\alpha_{os} = N_{rm} / N_{M}` where :math:`N_{rm}` is the
+          number of samples in the minority class after resampling and
+          :math:`N_{M}` is the number of samples in the majority class.
+
+            .. warning::
+               ``float`` is only available for **binary** classification. An
+               error is raised for multi-class classification.
+
+        - When ``str``, specify the class targeted by the resampling. The
+          number of samples in the different classes will be equalized.
+          Possible choices are:
+
+            ``'minority'``: resample only the minority class;
+
+            ``'not minority'``: resample all classes but the minority class;
+
+            ``'not majority'``: resample all classes but the majority class;
+
+            ``'all'``: resample all classes;
+
+            ``'auto'``: equivalent to ``'not majority'``.
+
+        - When ``dict``, the keys correspond to the targeted classes. The
+          values correspond to the desired number of samples for each targeted
+          class.
+
+        - When callable, function taking ``y`` and returns a ``dict``. The keys
+          correspond to the targeted classes. The values correspond to the
+          desired number of samples for each class.
+
+    shrink_factors : dict, default= 1 for every class
+        Dict of {classes: shrinkfactors} items, applied to
+        the gaussian kernels.
```

Comment:
`shrink_factors` could use some additional explanation.

The docstring for `_make_samples` mentions:

    "An optional shrink factor can be included, to compress/dilate the kernel."

This explanation would be better as part of this docstring.

Reply:
added additional description